src/facade/dragonfly_connection.cc

```diff
@@ -1016,10 +1029,13 @@ void Connection::DispatchFiber(util::FiberSocketBase* peer) {
     queue_backpressure_->ec.notify();
   }
 
-  cc_->conn_closing = true;
+  if (builder->GetError())
+    cc_->conn_closing = true;
 
   // Recycle messages even from disconnecting client to keep properly track of memory stats
   for (auto& msg : dispatch_q_) {
+    if (msg.IsIntrusive())
+      visit(dispatch_op, msg.handle);  // to not miss checkpoints
```

Comment:
maybe it is safer to add a fiber guard above here to make sure intrusive handlers do not preemt

Reply:
Will do, but in theory there shouldn't be a problem

Reply:
We are iterating the dispatch_q_ here, if we preempt we could insert dispatch_q_ from another fiber